src/pages/dashboard/components/EpisodeDetails.tsx

```diff
@@ -66,10 +66,38 @@ function EpisodeDetails({ episode, isInCollection = false, showDate = false }: P
     return `${((resumePosition.asMilliseconds() / duration.asMilliseconds()) * 100).toFixed(2)}%`;
   }, [episode.Duration, episode.ResumePosition]);
 
+  const anidbEpisodeAlias = (type: EpisodeTypeEnum, epTitle: string) => {
+    // TODO: This is a hack, we need update backend to make the EpisodeTypeEnum make more sense.
+    if (type === EpisodeTypeEnum.ThemeSong) {
+      if (epTitle.includes('Opening')) return 'OP';
+      if (epTitle.includes('Ending')) return 'ED';
+      return '';
+    }
+
+    // Prefixes for episode types base on https://wiki.anidb.net/Content:Episodes
+    switch (type) {
+      case EpisodeTypeEnum.OpeningSong:
+        return 'OP';
+      case EpisodeTypeEnum.EndingSong:
+        return 'ED';
+      case EpisodeTypeEnum.Special:
+      case EpisodeTypeEnum.Extra:
+        return 'S';
+      case EpisodeTypeEnum.Trailer:
+        return 'T';
+      case EpisodeTypeEnum.Other:
+        return 'O';
+      case EpisodeTypeEnum.Parody:
+        return 'P';
+      default:
+        return '';
+    }
+  };
+
   const airDate = useMemo(() => dayjs(episode.AirDate), [episode.AirDate]);
   const relativeTime = useMemo(() => airDate.calendar(null, CalendarConfig), [airDate]);
   const title = useMemo(
-    () => `${episode.Type === EpisodeTypeEnum.Normal ? '' : episode.Type[0]}${episode.Number} - ${episode.Title}`,
+    () => `${anidbEpisodeAlias(episode.Type, episode.Title)}${episode.Number} - ${episode.Title}`,
```

Comment:
If we want to do client-side OP/ED parsing w/variations (e.g. `OP1a`) then I recommend passing in the episode number to the alias function and creating the full string from within the function.